apps/champions/lib/champions/units.ex

```diff
@@ -207,6 +217,151 @@ defmodule Champions.Units do
       %CurrencyCost{currency_id: Currencies.get_currency_by_name!("Gems").id, amount: 50}
     ]
 
+  ##########
+  # Fusion #
+  ##########
+
+  @doc """
+  Consume a list of units that meet specific rank and character requirements based on the target
+  unit's rank in order to increase it.
+
+  Returns `{:ok, unit}` or `{:error, reason}`.
+  """
+  def fuse(user_id, unit_id, consumed_units_ids) do
```

Comment:
We should ensure that the unit_id is not inside the consumed_units_ids list since it should not count as a consume valid unit